toqito/states/mutually_unbiased_basis.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+from sympy import primerange, isprime
+
+from toqito.matrices import gen_pauli
+
+
+def mutually_unbiased_basis(dim: int) -> list[np.ndarray]:
+    """
+    Function to generate list of MUBs for a given dimension.
+    
+    Note that this function only works if the dimension provided is prime or a power of a prime. Otherwise, we don't
+    know how to generate general MUBs.
+
+    Examples
+    ========
+    
+    For the case of dimension 2, the three mutually unbiased bases are provided by:
+
+    .. math::
+        M_0 = \left\{|0\rangle, |1\rangle \right\}, \\
+        M_1 = \left\{\frac{|0\rangle + |1\rangle}{\sqrt{2}}, \frac{|0\rangle - |1\rangle}{\sqrt{2}}\right\}
+        M_2 = \left\{\frac{|0\rangle + i|1\rangle}{\sqrt{2}}, \frac{|0\rangle - i|1\rangle}{\sqrt{2}}\right\}
+
+    References
+    ==========
+    .. [WikMUB] Wikipedia: Mutually unbiased bases
+        https://en.wikipedia.org/wiki/Mutually_unbiased_bases
+    
```

Comment:
Same as what I said in `state_props/is_mutually_unbiased_basis`. 

I might be wrong but I don't think this reference has been cited in this function's docstring. So, using `docname in docnames` won't automatically generate a reference.

Reply:
Ah, let me add that citation. Good call!